scripts/docker/entrypoint.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+handle_term()
+{
+    echo "received TERM signal"
+    echo "stopping nginx-agent ..."
+    kill -TERM "${agent_pid}" 2>/dev/null
+    echo "stopping nginx ..."
+    kill -TERM "${nginx_pid}" 2>/dev/null
+}
+
+trap 'handle_term' TERM
+
+# Launch nginx
+echo "starting nginx ..."
+/usr/sbin/nginx -g "daemon off;" &
+
+nginx_pid=$!
+
+SECONDS=0
+
+while ! ps -ef | grep "nginx: master process" | grep -v grep; do 
+    if (( SECONDS > 5 )); then 
+        echo "couldn't find nginx master process"
+        exit 1
+    fi
+done
```

Comment:
Should `SECONDS` be incremented?

Reply:
No the `SECONDS` variable is a special built in variable in bash that increments itself as time passes.

From [bash man page](https://man7.org/linux/man-pages/man1/bash.1.html):

SECONDS

Each time this parameter is referenced, the number of seconds since shell invocation is returned. If a value is assigned to SECONDS, the value returned upon subsequent references is the number of seconds since the assignment plus the value assigned. If SECONDS is unset, it loses its special properties, even if it is subsequently reset.